src/workloads/query/QueryStats.yml

```diff
@@ -10,233 +10,179 @@ KeyWords:
-                [{a1: 1}, {a2: 1}, {^Object: {
-                  withNEntries: {^RandomInt: {min: 1, max: 350}},
-                  havingKeys: {^RandomString: {length: 10}},
-                  andValues: {$exists: false},
-                  duplicatedKeys: skip
-                }}]
-              }
-
-- Name: Quiesce
-  Type: QuiesceActor
-  Threads: 1
-  Database: *Database
-  Phases:
-    OnlyActiveInPhases:
-      Active: [5]
-      NopInPhasesUpTo: *MaxPhases
-      PhaseConfig:
-        Repeat: 1
-
-- Name: FindConcurrentlyWithQueryStatsWithHMAC
-  Type: CrudActor
-  Threads: 32
-  Database: *Database
-  Phases:
-    OnlyActiveInPhases:
-      Active: [6]
-      NopInPhasesUpTo: *MaxPhases
-      PhaseConfig:
-        Duration: 1 minute
-        Collection: *Collection
-        Operations:
-        - OperationName: find
-          OperationCommand:
-            Filter:
-              {$and:
-                [{a1: 1}, {a2: 1}, {^Object: {
-                  withNEntries: {^RandomInt: {min: 1, max: 350}},
-                  havingKeys: {^RandomString: {length: 10}},
-                  andValues: {$exists: false},
-                  duplicatedKeys: skip
-                }}]
-              }
-
-# QueryStats with HMAC.
-- ActorFromTemplate:
-    TemplateName: QueryStatsAggregation
-    TemplateParameters:
-      Name: "QueryStatsAggregationWithHMAC"
-      queryStatsParameters: {transformIdentifiers: {algorithm: "hmac-sha-256", hmacKey: *binDataGen}}
-      OnlyActiveInPhase: 6
-
-- Name: Quiesce
-  Type: QuiesceActor
-  Threads: 1
-  Database: *Database
-  Phases:
-    OnlyActiveInPhases:
-      Active: [7]
-      NopInPhasesUpTo: *MaxPhases
-      PhaseConfig:
-        Repeat: 1
-
-- Name: FindConcurrentlyWithQueryStatsWithoutHMAC
-  Type: CrudActor
-  Threads: 32
-  Database: *Database
-  Phases:
-    OnlyActiveInPhases:
-      Active: [8]
-      NopInPhasesUpTo: *MaxPhases
-      PhaseConfig:
-        Duration: 1 minute
-        Collection: *Collection
-        Operations:
-        - OperationName: find
-          OperationCommand:
-            Filter:
-              {$and:
-                [{a1: 1}, {a2: 1}, {^Object: {
-                  withNEntries: {^RandomInt: {min: 1, max: 350}},
-                  havingKeys: {^RandomString: {length: 10}},
-                  andValues: {$exists: false},
-                  duplicatedKeys: skip
-                }}]
-              }
-
-# QueryStats without HMAC.
-- ActorFromTemplate:
-    TemplateName: QueryStatsAggregation
-    TemplateParameters:
-      Name: "QueryStatsAggregationWithoutHMAC"
-      queryStatsParameters: {}
-      OnlyActiveInPhase: 8
+- LoadConfig:
+    Path: ../../phases/query/QueryStats.tmpl
+    Key: ClearCollection  # Drops the collection, only needs to happen once.
+    Parameters:
+      MaxPhase: *MaxPhase
+      Phase: 0
+- LoadConfig:
+    Path: ../../phases/query/QueryStats.tmpl
+    Key: InsertData  # Repopulates the collection, only needs to happen once.
```

Comment:
Can we make the data format schema live here where we have the query format schema also? Taking the documents as a template argument to the insertion actor?

Or did you already try this and it didn't work for some reason?

Reply:
Actually - looking closer - more of the queries are in the template file, so it's fine that the data document is next to those queries - do the "fixed" and "random" payloads need to live in this file? Curious to hear your thought process. I am just trying to cut down on the need to cross-reference the two and have them both open.